src/CustomFunctionsTrait.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace RebelCode\CommandInvoker;
+
+use Traversable;
+use InvalidArgumentException;
+use OutOfRangeException;
+use DomainException;
+use Exception as RootException;
+use Dhii\Util\String\StringableInterface as Stringable;
+
+/**
+ * Functionality for custom functions.
+ *
+ * @since [*next-version*]
+ */
+trait CustomFunctionsTrait
+{
+    /**
+     * The map of codes to functions.
+     *
+     * @since [*next-version*]
+     *
+     * @var callable[]
+     */
+    protected $customFunctions;
+
+    /**
+     * Registers a custom function.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $code     The code to register the function with.
+     * @param callable          $function The function to register.
+     *
+     * @throws InvalidArgumentException If function is not a callable.
+     */
+    protected function _registerCustomFunction($code, $function)
```

Comment:
I thought there may be different objects that implement a different interface which makes them invocable via a different mechanism. Then again, maybe if it's invocable, it MUST declare `__invoke()`. Thoughts?

Reply:
Yes, and if it declares `__invoke()` then it qualifies as a `callable` as well.

Reply:
Yea, I know, but thought that maybe we could make this function even more flexible. But maybe tighter is better.

Reply:
If there's room for flexibility, I'm all for it. Like if `null` is accepted, or other things qualify as functions other than `callable` instances. Not sure what other things would qualify, since PHP already has a wide range of things that it considers to be `callable`, such as strings and arrays that represent functions or methods. lol

Reply:
An invoker could be considered callable. That's the left argument. The right argument is that if it is considered invokable, it can make everything else understand that by implementing `__invoke()`.

Reply:
Right argument FTW 💪